### PR TITLE
9555 sarhamam replace bytes with bytes

### PR DESCRIPTION
**Pull request by Sarhamam**
When replacing bytes, encode `os.linesep` which is str, and prefix `\r\n` with b.

### CODE REVIEW

**Comment by twm** [comment]
Hi @Sarhamam, this looks like a reasonable fix but you'll have to [add some sort of test coverage of this function](https://twistedmatrix.com/trac/wiki/ReviewProcess#Authors:Thingsyourbranchorpatchmustcontain) before we can merge it.

File: src/twisted/protocols/ftp.py [hunk 1]
```diff
@@ -667,7 +667,7 @@ def __init__(self, cons):
             self.write = cons.write
```
Comment (graingert) [inline src/twisted/protocols/ftp.py:667]: This conditional is wrong in the same way too

File: src/twisted/protocols/ftp.py [hunk 2]
```diff
@@ -667,7 +667,7 @@ def __init__(self, cons):
...
 
     def write(self, bytes):
-        return self.cons.write(bytes.replace(os.linesep, "\r\n"))
+        return self.cons.write(bytes.replace(os.linesep.encode(), b"\r\n"))
```
Comment (graingert) [inline src/twisted/protocols/ftp.py:670]: Probably worth setting
```
_LINESEP_BYTES = os.linesep.encode()
```
somewhere

File: src/twisted/protocols/ftp.py (hunk 1 above)
Comment (Sarhamam) [reply src/twisted/protocols/ftp.py:667]: I dont think this conditional is wrong, `os.linesep` return str type, and so is "\r\n", no need to convert to bytes here.

File: src/twisted/protocols/ftp.py (hunk 2 above)
Comment (Sarhamam) [reply src/twisted/protocols/ftp.py:670]: Done at the top of the file, along with _CRLF_BYTES constant.

**Comment by Sarhamam** [comment]
Having problems with fork - will create new PR once ready.

**Pull request closed without merging**